ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BVC+WKDownloadDelegate.swift

```diff
@@ -3,205 +3,127 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
+import BraveCore
 import BraveShared
 import Foundation
+import OSLog
 import PassKit
 import Shared
 import WebKit
 
-extension BrowserViewController: WKDownloadDelegate {
+extension BrowserViewController: CWVDownloadTaskDelegate {
+  fileprivate func uniqueDownloadPathForFilename(_ filename: String) async throws -> URL {
+    let downloadsPath = try await AsyncFileManager.default.downloadsPath()
+    let basePath = downloadsPath.appending(path: filename)
+    let fileExtension = basePath.pathExtension
+    let filenameWithoutExtension =
+      !fileExtension.isEmpty ? String(filename.dropLast(fileExtension.count + 1)) : filename
 
-  public func download(
-    _ download: WKDownload,
-    decideDestinationUsing response: URLResponse,
-    suggestedFilename: String
-  ) async -> URL? {
+    var proposedPath = basePath
+    var count = 0
 
-    if let httpResponse = response as? HTTPURLResponse {
-      if httpResponse.mimeType != MIMEType.passbook {
-        let shouldDownload = await downloadAlert(
-          download,
-          response: response,
-          suggestedFileName: suggestedFilename
-        )
-        if !shouldDownload {
-          return nil
-        }
-      }
-    }
+    while await AsyncFileManager.default.fileExists(atPath: proposedPath.path) {
+      count += 1
 
-    let temporaryDir = NSTemporaryDirectory()
-    let fileName = temporaryDir + "/" + suggestedFilename
-    let url = URL(fileURLWithPath: fileName)
-
-    // WKDownload will fail with a -3000 error code if the file already exists at the given path
-    if await AsyncFileManager.default.fileExists(atPath: url.path(percentEncoded: false)) {
-      try? await AsyncFileManager.default.removeItem(at: url)
+      let proposedFilenameWithoutExtension = "\(filenameWithoutExtension) (\(count))"
+      proposedPath = downloadsPath.appending(path: proposedFilenameWithoutExtension)
+        .appending(path: fileExtension)
```

Comment:
Needs to be fixed same way as https://github.com/brave/brave-core/pull/26865/files